resources/metadata/protecode.yaml

```diff
@@ -0,0 +1,229 @@
+spec:
+  inputs:
+    params:
+    - name: protecodeExcludeCVEs
+      type: string
+      description: 'DEPRECATED: Do use triaging within the Protecode UI instead'
+      mandatory: false
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: []
+    - name: protecodeFailOnSevereVulnerabilities
+      type: bool
+      description: Whether to fail the job on severe vulnerabilties or not
+      mandatory: false
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: true
+    - name: scanImage
+      aliases:
+      - name: dockerImage
+      type: string
+      description: The reference to the docker image to scan with Protecode
+      mandatory: false
+      resourceRef:
+        - name: commonPipelineEnvironment
+          param: container/imageNameTag
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: null
+    - name: dockerRegistryUrl
+      type: string
+      description: The reference to the docker registry to scan with Protecode
+      mandatory: false
+      resourceRef:
+        - name: commonPipelineEnvironment
+          param: container/registryUrl
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: null
+    - name: cleanupMode
+      type: string
+      description: Decides which parts are removed from the Protecode backend after the scan
+      mandatory: false
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: binary
+    - name: filePath
+      type: string
+      description: The path to the file from local workspace to scan with Protecode
+      mandatory: false
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: null
+    - name: includeLayers
+      type: bool
+      description: Flag if the docker layers should be included
+      mandatory: false
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: false
+    - name: addSideBarLink
+      type: bool
+      description: Whether to create a side bar link pointing to the report produced by Protecode or not
+      mandatory: false
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: true
+    - name: verbose
+      type: bool
+      description: Whether to log verbose information or not
+      mandatory: false
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: false
+    - name: protecodeTimeoutMinutes
+      type: string
+      description: The timeout to wait for the scan to finish
+      mandatory: false
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: 60
+    - name: protecodeServerUrl
+      type: string
+      description: The URL to the Protecode backend
+      mandatory: false
+      scope:
+      - GENERAL
```

Comment:
not sure if we need this as `GENERAL` parameter?

If not, we could also remove `protecode` from the parameter name